N24DCCN017/buoi5/bai10.cpp

```diff
@@ -0,0 +1,39 @@
+#include<iostream>
+#include<math.h>
+using namespace std;
+bool kiemtra(int n){
+    bool kt=true;
+    if(n==2) kt=true;
+    else {
+    for(int i=2;i<=int(sqrt(n));i++){
+        if(n%i==0) {
+            kt=false; 
+            break;
+            }
+    }
+    }
+    return kt;
+}
+int main(){
+    int x;
+    cout<<"nhap 1 so can phan tich: ";cin>>x;
+    for(int i=2;i<=int(sqrt(x));i++){
+        int mu=1;
+        if(kiemtra(i)){
+            if(!(x%i)){
+                cout<<i;
+                x/=i;
+            }
+            if(!(x%i)){
+                cout<<"^";
+                while(!(x%i)){
+                    mu+=1;
+                    x/=i;
+                }
+                cout<<mu;
+            }
+        if(x!=1)cout<<" x ";
+        }
+    }
+
+}
```

Comment:
K hiểu thuật toán em làm lắm. Ý tưởng bài này là chạy biến i để duyệt từng cơ số. Nếu n chia hết cho i và i là snt thì mũ của cơ số i sẽ cộng lên, giảm n và tiếp tục chia cho i. nếu n còn chia hết cho i thì tiếp tục cộng biến mũ, không chia hết thì tăng i lên để tìm cơ số tiếp theo